File: baekjoon/10000/10866.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    cin.tie(NULL), cout.tie(NULL);
    ios::sync_with_stdio(false);

    deque<int> dq;
    int n, x;
    string order;
    cin >> n;
    while (n--) {
        cin >> order;
        if (order == "push_front") {
            cin >> x;
            dq.push_front(x);
        }
        else if (order == "push_back") {
            cin >> x;
            dq.push_back(x);
        }
        else if (order == "pop_front") {
            if (dq.empty()) cout << "-1\n";
            else {
                cout << dq.front() << '\n';
                dq.pop_front();
            }
        }
        else if (order == "pop_back") {
            if (dq.empty()) cout << "-1\n";
            else {
                cout << dq.back() << '\n';
                dq.pop_back();
            }
        }
        else if (order == "size")
            cout << dq.size() << '\n';
        else if (order == "empty")
            cout << dq.empty() << '\n';
        else if (order == "front") {
            if (dq.empty()) cout << "-1\n";
            else cout << dq.front() << '\n';
        }
        else{
            if (dq.empty()) cout << "-1\n";
            else cout << dq.back() << '\n';
        }
    }
    return 0;
}

File: baekjoon/10000/10866.rs
use std::io::*;

struct Scanner {
    it: std::str::SplitAsciiWhitespace<'static>,
}

impl Scanner {
    fn new() -> Self {
        let mut s = String::new();
        stdin().read_to_string(&mut s).ok();
        Self { it: s.leak().split_ascii_whitespace() }
    }
    fn read<T: std::str::FromStr>(&mut self) -> T {
        self.it.next().unwrap().parse::<T>().ok().unwrap()
    }
    fn raw(&mut self) -> &str {
        self.it.next().unwrap()
    }
}

fn main() {
    let mut sc = Scanner::new();
    let mut bw = BufWriter::new(stdout().lock());
    let n = sc.read::<usize>();
    let mut dq = std::collections::VecDeque::new();
    for _ in 0..n {
        match sc.raw() {
            "push_front" => dq.push_front(sc.read::<i32>()),
            "push_back" => dq.push_back(sc.read::<i32>()),
            "pop_front" => writeln!(bw, "{}", dq.pop_front().unwrap_or(-1)).unwrap(),
            "pop_back" => writeln!(bw, "{}", dq.pop_back().unwrap_or(-1)).unwrap(),
            "size" => writeln!(bw, "{}", dq.len()).unwrap(),
            "empty" => writeln!(bw, "{}", dq.is_empty() as usize).unwrap(),
            "front" => writeln!(bw, "{}", dq.front().unwrap_or(&-1)).unwrap(),
            "back" => writeln!(bw, "{}", dq.back().unwrap_or(&-1)).unwrap(),
            _ => unreachable!(),
        }
    }
}
